### PR TITLE
migrate to null safety

### PR DESCRIPTION
**Pull request by tshedor**
@jonsamwell This is a sibling PR to jonsamwell/dart_gherkin#30 and requires it. The updates to /example/android address [the deprecated Android API message](https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects).

### CODE REVIEW

**Comment by jonsamwell** [comment]
@tshedor aiming to get null-safety sorted out next week, thanks for your effort here!

File: lib/src/flutter/flutter_test_configuration.dart [hunk 1]
```diff
@@ -41,7 +40,7 @@ class FlutterTestConfiguration extends TestConfiguration {
     String targetAppPath = 'test_driver/app.dart',
   }) {
     return FlutterTestConfiguration()
-      ..features = [Glob(featurePath)]
+      ..features = [RegExp(featurePath)]
```
Comment (marcin-jelenski) [inline lib/src/flutter/flutter_test_configuration.dart:43]: When switching to `RegExp`, the `featurePath` value from `DEFAULT` configuration has to be changed:

```dart
String featurePath = 'test_driver\/features\/.*.feature',
```

Comment (tshedor) [reply lib/src/flutter/flutter_test_configuration.dart:43]: I think this would be a separate PR as it's not specifically related to null safety, right?

Comment (tshedor) [reply lib/src/flutter/flutter_test_configuration.dart:43]: Ah, no, nevermind you're right, since `RegExp` was introduced in gherkin 2.0

**Comment by dened** [comment]
Hi, is any update about this PR?

**Comment by jonsamwell** [comment]
@tshedor thanks for this, again it was a great reference for null-safety migration.  Unfortunately this migrates the current master branch and diverges too far from the branch containing the work to enable the use of `integration_test` so I have begun to migration that branch (help and review appreciated) see https://github.com/jonsamwell/flutter_gherkin/pull/125.

Unfortunately I don't think this library can achieve null-safety yet neither flutter_driver and intergration_test are null safe themselves yet (at least in the flutter stable branch).

**Comment by jonsamwell** [comment]
@tshedor if you resolve the conflicts with this PR now after the latest round of chance I think we'll be good to go

**Comment by tshedor** [comment]
@jonsamwell conflicts resolved; sorry I was out of internet access for a few days.

**Comment by tshedor** [comment]
Hey @jonsamwell do you need anything else on this?

**Comment by tshedor** [comment]
@jonsamwell gentle nudge on this

**Pull request merged**